lib/layer.js

```diff
@@ -90,8 +91,9 @@ Layer.prototype.handle_request = function handle(req, res, next) {
   }
 
   try {
-    fn(req, res, next)
+    fn(req, res, once(next))
   } catch (err) {
+    if (err === 'once') { throw new Error('cannot call `next` more than once') }
```

Comment:
We definitely do not want to ever purposely throw something that is un-catachable in some way.